### PR TITLE
fozzie-components@v2.17.0 - Consolidate data-test-ids and data-js-test attributes #globalconfig

### PR DESCRIPTION
**Pull request by LTurns**
### Changed

- Consolidated test ids - replaced `data-js-test` attribute with `data-test-id` where possible, to be shared by unit and component tests. The only test ids in the repo should now be either 'data-test-id' or 'data-theme-{component-name}', where required.

### CODE REVIEW

File: packages/f-header/test-utils/component-objects/f-header.component.js [hunk 1]
```diff
@@ -9,9 +9,11 @@ exports.clickLoginLink = () => {
 }
 
 exports.clickOffersLink = () => {
-    offersLink().click();
+    offersLinkMobile().click();
 }
 
+//do one for offerslinkdesktop too
```
Comment (xander-marjoram) [inline packages/f-header/test-utils/component-objects/f-header.component.js]: @LTurns Do we have a ticket for this or is this still a WIP?

File: packages/f-registration/test-utils/component-objects/f-registration.component.js [hunk 2]
```diff
@@ -10,19 +10,19 @@ const privacyPolicyLink = () => registrationComponent().$('[data-test-id="privac
 const cookiesPolicyLink = () => registrationComponent().$('[data-test-id="cookies-policy-link"]');
 
 // Validation errors
-const firstNameEmptyError = () => $('[data-test-title="error-first-name-empty"]');
-const firstNameMaxLengthError = () => $('[data-test-title="error-first-name-maxlength"]');
+const firstNameEmptyError = () => $('[data-test-id="error-first-name-empty"]');
+const firstNameMaxLengthError = () => $('[data-test-id="error-first-name-maxlength"]');
 const firstNameInvalidError = () => $('[data-test-title="error-first-name-invalid"]');
```
Comment (xander-marjoram) [inline packages/f-registration/test-utils/component-objects/f-registration.component.js]: I think you missed this one 😄

File: packages/f-header/test-utils/component-objects/f-header.component.js (hunk 1 above)
Comment (LTurns) [reply packages/f-header/test-utils/component-objects/f-header.component.js]: @xander-marjoram think I'll quickly do that now actually - should only be one small test. Thanks for reminding me!

File: packages/f-registration/test-utils/component-objects/f-registration.component.js (hunk 2 above)
Comment (LTurns) [reply packages/f-registration/test-utils/component-objects/f-registration.component.js]: whoops, thank youu!

**Comment by ashleynolan** [comment]
I've added `#globalconfig` to your title so that it stops asking for CHANGELOG entries at package level

**Comment by LTurns** [comment]
> I've added `#globalconfig` to your title so that it stops asking for CHANGELOG entries at package level

thanks Ash, will remember to do that next time! :)

File: packages/f-header/test-utils/component-objects/f-header.component.js [hunk 3]
```diff
@@ -1,5 +1,6 @@
 const loginLink = () => $('[data-test-id="login-link"]');
-const offersLink = () => $('[data-test-id="offers-link]"');
+const offersLinkMobile = () => $('[data-test-id="offers-link-mobile]"');
+const offersLinkDesktop = () => $('[data-test-id="offers-link-desktop]"');
```
Comment (siggerzz) [inline packages/f-header/test-utils/component-objects/f-header.component.js]: Is there a way we can consolidate this into a single selector that can be used for both mobile and desktop?

**Review by siggerzz — CHANGES_REQUESTED** [review]
See comments!

File: packages/f-header/test-utils/component-objects/f-header.component.js (hunk 3 above)
Comment (xander-marjoram) [reply packages/f-header/test-utils/component-objects/f-header.component.js]: Could we have two data-test-ids on each link, one "offers-link-mobile/desktop" for the mobile/desktop tests and one "offers-link" for generic use?

Comment (minrwhite) [reply packages/f-header/test-utils/component-objects/f-header.component.js]: Wouldn't the attribute starts-with selector `$('[data-test-id^="offers-link"]')` suffice for [the case where we need to match either]? Also the closing quotes and square-parens are in the wrong order above: 
```suggestion
const offersLinkDesktop = () => $('[data-test-id="offers-link-desktop"]');
```

File: packages/f-header/test-utils/component-objects/f-header.component.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
 const loginLink = () => $('[data-test-id="login-link"]');
-const offersLink = () => $('[data-test-id="offers-link]"');
+const offersLinkMobile = () => $('[data-test-id="offers-link-mobile]"');
```
Comment (minrwhite) [inline packages/f-header/test-utils/component-objects/f-header.component.js]: This line also needs that change:
```suggestion
const offersLinkMobile = () => $('[data-test-id="offers-link-mobile"]');
```

**Review by minrwhite — APPROVED** [review]

**Review by siggerzz — APPROVED** [review]

**Pull request merged**